daemon/algod/api/server/v2/handlers.go

```diff
@@ -584,10 +584,10 @@ func (v2 *Handlers) getPendingTransactions(ctx echo.Context, max *uint64, format
 	}
 
 	// Convert transactions to msgp / json strings
-	txnArray := make([]transactions.SignedTxn, 0)
-	for _, txn := range txns {
+	topTxns := make([]transactions.SignedTxn, 0)
+	for _, txn := range txnPool {
 		// break out if we've reached the max number of transactions
-		if max != nil && uint64(len(txnArray)) >= *max {
+		if max != nil && *max != 0 && uint64(len(topTxns)) >= *max {
```

Comment:
Nit : evaluate the max variable outside the loop, and initialize another variable to the desired limit.
If no limit, use maxint.

This would simplify the in-loop predicate.